File: test_examples/CW1-s2/tree.py
"""CW1-s2 working solution in multiple files"""

from internals import find_internal

class Tree:
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None

    def insert(self, value):
        if value == self.value:
            return False
        if value < self.value:
            if self.left is None:
                self.left = Tree(value)
                return True
            else:
                return self.left.insert(value)
        if value > self.value:
            if self.right is None:
                self.right = Tree(value)
                return True
            else:
                return self.right.insert(value)

    def find(self, needle):
        return find_internal(self, needle, 0)
    

File: test_examples/CW3-s3/Tree.java
public class Tree {

	private Tree Left;
	private Tree Right;	
	private int value;

	public Tree(int value){
		this.value = value;
	}

	public Tree Smaller(){
		return this.Right;
	}

	public Tree Greater(){
		return this.Right;
	}
	
	public int Val(){
		return value;
	}
	
	public boolean Insert(int newval){
		if(newval<this.Val()){
			Tree smaller = this.Smaller();
			if(smaller==null){
				this.Left = new Tree(newval);
				return true;
			} else {
				return this.Left.Insert(newval);
			}
		}
		if(newval>this.Val()){
			Tree greater = this.Greater();
			if(greater==null){
				this.Right = new Tree(newval);
				return true;
			} else {
				return greater.Insert(newval);
			}
		}
		return false;
	}
}
